packages/react-components/react-motion-components-preview/library/src/components/Fade/Fade.test.tsx

```diff
@@ -0,0 +1,130 @@
+
+  beforeAll(() => {
+    HTMLElement.prototype.animate = () => mockAnimation;
+  });
+
+  beforeEach(() => {
+    animateSpy = jest.spyOn(HTMLElement.prototype, 'animate').mockImplementation(() => mockAnimation);
+  });
+
+  afterEach(() => {
+    animateSpy.mockRestore();
+  });
+
+  afterAll(() => {
+    delete (HTMLElement.prototype as unknown as { animate?: Function }).animate;
+  });
+
+  it('should render Fade with correct opacity keyframes, duration and easing (visible=false -> true -> false)', () => {
+    const { rerender } = render(<Fade visible={false}>{testElement}</Fade>);
+
+    // Testing fade in motion
+    rerender(<Fade visible={true}>{testElement}</Fade>);
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationNormal, easing: motionTokens.curveEasyEase }),
+    );
+
+    // Testing fade out motion
+    rerender(<Fade visible={false}>{testElement}</Fade>);
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 1 }, { opacity: 0 }],
+      expect.objectContaining({ duration: motionTokens.durationNormal, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render Snappy variant of Fade component with correct opacity keyframes, duration and easing', () => {
+    const { rerender } = render(<FadeSnappy visible={false}>{testElement}</FadeSnappy>);
+
+    rerender(<FadeSnappy visible={true}>{testElement}</FadeSnappy>);
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationFast, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render Exaggerated variant of Fade component with correct opacity keyframes, duration and easing', () => {
+    const { rerender } = render(<FadeExaggerated visible={false}>{testElement}</FadeExaggerated>);
+
+    rerender(<FadeExaggerated visible={true}>{testElement}</FadeExaggerated>);
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationGentle, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render child element with overriding styles and still apply Fade animations', () => {
+    const styledElement = (
+      <div data-testid="fade-content" style={{ opacity: 0.5 }}>
+        Test
+      </div>
+    );
+
+    const { rerender } = render(<Fade visible={false}>{styledElement}</Fade>);
+
+    rerender(<Fade visible={true}>{styledElement}</Fade>);
+
+    expect(animateSpy).toHaveBeenCalledWith(
+      [{ opacity: 0 }, { opacity: 1 }],
+      expect.objectContaining({ duration: motionTokens.durationNormal, easing: motionTokens.curveEasyEase }),
+    );
+  });
+
+  it('should render Fade with parent styles that override Fade opacity and still apply Fade animations (visible=false -> true -> false)', async () => {
```

Comment:
Similarly to my comment on the test `it('should render child element with overriding styles and still apply Fade animations'`, this kind of style test is not possible in Jest as-is because JSDOM doesn't support Web Animations API.